### PR TITLE
feat(account-lib): add support for ecdsa keyshare generation tss

### PR DESCRIPTION
**Pull request by sachushajiabraham**
this change add support for ecdsa signature key share generation for TSS

TICKET: BG-48875

### CODE REVIEW

File: modules/account-lib/src/mpc/tss/ecdsa/ecdsa.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import Shamir from '../../shamir';
+import { bigIntToHex } from '../../util';
+import { NShare, PShare, KeyShare, KeyCombined } from './types';
+
+export default class Ecdsa {
+  static curve: Secp256k1Curve = new Secp256k1Curve();
+  static shamir: Shamir = new Shamir(Ecdsa.curve);
+  /**
+   * Generate shares for player at index and split keys `(t,n)` ways.
+   * @param {Number} index Player index
```
Comment (wesgraham) [inline modules/account-lib/src/mpc/tss/ecdsa/ecdsa.ts:12]: where does the term "player" come from? can we use "participant" in our naming to be consistent with docs/other implementations

**Review by wesgraham — DISMISSED** [review]

File: modules/account-lib/src/mpc/tss/ecdsa/ecdsa.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import Shamir from '../../shamir';
+import { bigIntToHex } from '../../util';
+import { NShare, PShare, KeyShare, KeyCombined } from './types';
+
+export default class Ecdsa {
+  static curve: Secp256k1Curve = new Secp256k1Curve();
+  static shamir: Shamir = new Shamir(Ecdsa.curve);
+  /**
+   * Generate shares for participant at index and split keys `(threshold,numShares)` ways.
+   * @param {Number} index participant index
```
Comment (mmcshinsky-bitgo) [inline modules/account-lib/src/mpc/tss/ecdsa/ecdsa.ts:12]: Inconsistent doc casing with method params.

**Review by mmcshinsky-bitgo — CHANGES_REQUESTED** [review]

File: modules/account-lib/src/mpc/util.ts [hunk 3]
```diff
@@ -30,6 +30,10 @@ export function bigIntToBufferBE(n: bigint, bytes?: number): Buffer {
   return buf;
 }
 
+export function bigIntToHex(a: bigint): string {
```
[the remark this answers is not in the document]
Comment (sachushajiabraham) [reply modules/account-lib/src/mpc/util.ts]: Yup. Updated.

**Review by johnoliverdriscoll — DISMISSED** [review]

**Comment by OttoAllmendinger** [comment]
Why was @BitGo/btc-team added as reviewer? Removed it.

**Review by ducnguyen-bitgo — APPROVED** [review]
Lgtm.

**Comment by sachushajiabraham** [comment]
@mmcshinsky-bitgo can you kindly re check this PR

**Review by mmcshinsky-bitgo — APPROVED** [review]

File: modules/sdk-core/src/account-lib/mpc/index.ts [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
-import { KeyShare } from './tss/eddsa/types';
+import { EDDSA } from './tss';
+
+export { Ecdsa, ECDSA, Eddsa, EDDSA } from './tss';
 
 export * from './curves';
 export * from './util';
 
-export { Eddsa, Ed25519BIP32, HDTree, KeyShare };
+type KeyShare = EDDSA.KeyShare;
```
Comment (olibeyene) [inline modules/sdk-core/src/account-lib/mpc/index.ts:9]: can ecdsa and eddsa share KeyShare type?

Comment (sachushajiabraham) [reply modules/sdk-core/src/account-lib/mpc/index.ts:9]: Nope , Left Keyshare export  like this to be backward compatible.

**Pull request merged**